save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt

```diff
@@ -87,9 +88,15 @@ class TestExecutionController(private val testExecutionService: TestExecutionSer
     ): Mono<List<TestSuiteExecutionStatisticDto>> =
             justOrNotFound(executionService.findExecution(executionId)).filterWhen {
                 projectPermissionEvaluator.checkPermissions(authentication, it, Permission.READ)
-            }.map {
-                testExecutionService.getTestExecutions(executionId, page, size).groupBy { it.test.testSuite.name }.map { (testSuiteName, testExecutions) ->
-                    TestSuiteExecutionStatisticDto(testSuiteName, testExecutions.count(), testExecutions.count { it.status == status }, status)
+            }.mapNotNull {
+                if (page == null || size == null) {
+                    testExecutionService.getTestExecutions(executionId).groupBy { it.test.testSuite.name }.map { (testSuiteName, testExecutions) ->
```

Comment:
In the future we should also perform more calculations here on DB, because number of test executions may become very large